### PR TITLE
PROCESS0UT-35 - Invoice challenge indicator field

### PR DESCRIPTION
**Pull request by roshan-gorasia-cko**
Add instructions in regards to the new `challenge_indicator` field on the invoice

NOTE: Only to be merged in after the `sca_exemption_reason` PR is merged in first

### CODE REVIEW

File: source/refs/content/payments/_invoices.md.erb [hunk 1]
```diff
@@ -27,6 +27,7 @@ using the ProcessOut API. To start using invoices, [let's create one](#create-an
...
 <span class="console-font">**metadata**</span><br />[Metadata](#metadata)<br />*dictionary* | Context related to the invoice, key-value pair (string - string)
 <span class="console-font">**exemption_reason_3ds2**</span><br />*string* | Optional reason for requesting 3DS2 not to be invoked (Note: Must also be supported by the PSP. Please contact our support to know more!)<br /><span style="opacity: 0.5">Allowed values are: "low-value", "trusted-beneficiary" or "transaction-risk-analysis"</span>
+<span class="console-font">**challenge_indicator**</span><br />*string* | Optional challenge indicator field when requesting 3DS2 (Note: Must also be supported by the PSP. Please contact our support to know more!)<br /><span style="opacity: 0.5">Allowed values are: "no-preference", "no-challenge-requested", "challenge-requested", "challenge-requested-mandate", "no-challenge-requested-tra-performed", "no-challenge-requested-data-share-only", "no-challenge-requested-sca-performed", "no-challenge-requested-whitelist-exemption", "challenge-requested-whitelist-prompt" or "cb-scoring"</span>
```
Comment (roshan-gorasia-cko) [inline source/refs/content/payments/_invoices.md.erb:30]: Hey @pierre-lafanechaire-cko, myself and @matt-ridgway-cko wanted to get your input on something!

Matt raised an interesting point where the merchant may be confused between `exemption_reason_3ds2` (SCA exemption) and `challenge_indicator` as they similar sounding functionality etc.

Matt suggested to conjoin the two to something along the lines of `sca_challenge` which holds the challenge indicator fields, in addition to 3 more that refer the to SCA exemption: `no-challenge-requested-low-value`,`no-challenge-requested-tra-performed` and `no-challenge-requested-trusted-beneficiary`

I'm happy to re-work to conjoin the two if necessary! 

My POV when working on this task was that SCA exemptions and challenge indicator are two different functionalities in a sense that SCA exemptions (if the issuer accepts) will not trigger a 3DS2 flow at all, and challenge indicator is only for the 3DS2 challenge window. Keeping it separate at the time made sense as I had prior understanding of this, but a merchant without knowing too much around these fields may see it as being very similar. 

TL;DR - Join the two into `sca_challenge` (will have to notify if any merchants already using the `exemption_reason_3ds2` field) or leave as is with 2 separated fields

Comment (pierre-lafanechaire-cko) [reply source/refs/content/payments/_invoices.md.erb:30]: Hey Roshan,
Your point makes sense as the outcome may appear the same for both values.
However, their purposes is different:
- Exemption: `Request an SCA exemption for the transaction. The customer’s bank has the final say on whether or not it applies.`
- Challenge Indicator : `Indicate your preference for whether or not a 3DS challenge should be performed. The customer’s bank has the final say on whether the customer is challenged.`

Taken from the [Checkout.com SCA compliance guide](https://docs.checkout.com/risk-management/sca-compliance-guide#SCAcomplianceguide-PossibleSCAexemptionsexemptions)

Merchants can use both values to indicate their choice. With one field, it would not be possible.
I prefer to leave it as 2 separated fields.

Comment (roshan-gorasia-cko) [reply source/refs/content/payments/_invoices.md.erb:30]: Indeed, that was my thinking behind this. @matt-ridgway-cko Happy to leave it as 2 separate fields?

Comment (matt-ridgway-cko) [reply source/refs/content/payments/_invoices.md.erb:30]: I guess my thought process is; what if a merchant sets the SCA exemption (for example to "low-value"), but sets the 3DS2 challenge indicator to "challenge-requested". Given that 3DS2 is a way to implement SCA, this seems like an invalid choice. Does this open odd scenarios with regards to liability shift and will we need to add additional validation later?

The challenge indicator, seems to have preferences for both requesting 3DS2 to be enabled and disabled. It almost feels like a superset of the options we know exist for SCA exemptions. My suggestion was to combine the fields and then map the values internally only sending through options we know are valid for each integration, keeping our API agnostic.

**Comment by roshan-gorasia-cko** [comment]
To summarise discussions:
- `exemption_reason_3ds2` has been renamed to `sca_exemption_reason` - The API allows sending the old field for now, going forward merchants should use `sca_exemption_reason`.
- `challenge_indicator` field has been left as is

**Review by matt-ridgway-cko — APPROVED** [review]
